chia/wallet/wallet_node.py

```diff
@@ -605,27 +605,35 @@ async def receive_state_from_peer(
         target_concurrent_tasks: int = 20
         num_concurrent_tasks: int = 0
 
-        async def receive_and_validate(inner_state: CoinState, inner_idx: int):
+        async def receive_and_validate(inner_states: List[CoinState], inner_idx_start: int):
             try:
                 assert self.validation_semaphore is not None
                 async with self.validation_semaphore:
                     assert self.wallet_state_manager is not None
                     if header_hash is not None:
                         assert height is not None
-                        self.add_state_to_race_cache(header_hash, height, inner_state)
-                        self.log.info(f"Added to race cache: {height}, {inner_state}")
+                        for inner_state in inner_states:
+                            self.add_state_to_race_cache(header_hash, height, inner_state)
+                            self.log.info(f"Added to race cache: {height}, {inner_state}")
                     if trusted:
-                        valid = True
+                        valid_states = inner_states
                     else:
-                        valid = await self.validate_received_state_from_peer(inner_state, peer, cache, fork_height)
-                    if valid:
-                        self.log.info(f"new coin state received ({inner_idx + 1} / {len(items)})")
+                        valid_states = [
+                            inner_state
+                            for inner_state in inner_states
+                            if await self.validate_received_state_from_peer(inner_state, peer, cache, fork_height)
+                        ]
+                    if len(valid_states) > 0:
```

Comment:
Could do `== 0` with an early return and save indentation and people scanning around to make sure they know how that case is handled.  Certainly equivalent though.